crates/uv-cli/src/lib.rs

```diff
@@ -2959,7 +2959,7 @@ pub struct SyncArgs {
     /// `tool.uv.conflicts`, uv will report an error.
     ///
     /// May be provided multiple times.
-    #[arg(long, conflicts_with("only_group"))]
+    #[arg(long, conflicts_with_all = ["only_group", "only_dev"])]
     pub group: Vec<GroupName>,
```

Comment:
Hm, this is tough because `--group dev` and `--only-dev` seems fine, similarly for `--group foo` and `--only-group foo`. `--only-group foo` and `--group bar` seems like an error as does `--group foo` and `--only-dev`.